specs/lightclient/beacon-chain.md

```diff
@@ -526,15 +526,20 @@ def process_rewards_and_penalties(state: BeaconState) -> None:
         return
     flag_deltas = [get_flag_deltas(state, flag, numerator) for (flag, numerator) in get_flags_and_numerators()]
     deltas = flag_deltas + [get_inactivity_penalty_deltas(state)]
-    for (rewards, penalties) in deltas:
+    for (rewards, penalty) in deltas:
         for index in range(len(state.validators)):
             increase_balance(state, ValidatorIndex(index), rewards[index])
-            decrease_balance(state, ValidatorIndex(index), penalties[index])
+        # Bounds-friendly expansion for `denom *= (1 + penalty // GWEI_PER_ETH)`
+        state.balance_denominator = (
+            state.balance_denominator
+            + penalty
+            + (state.balance_denominator - GWEI_PER_ETH) * penalty // GWEI_PER_ETH
+        )
```

Comment:
((porting previous comments to here))
https://github.com/ethereum/eth2.0-specs/pull/2150/files#r550256452

@dankrad:

```suggestion
        state.balance_denominator = state.balance_denominator + state.balance_denominator * penalty // GWEI_PER_ETH
```